src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -533,6 +533,29 @@ void EncryptedFileMapping::write_page(size_t local_page_ndx) noexcept
         m_chunk_dont_scan[chunk_ndx] = 0;
 }
 
+void EncryptedFileMapping::write_and_update_all(size_t local_page_ndx, size_t begin_offset, size_t end_offset) noexcept
+{
+    // Go through all other mappings of this file and copy changes into those mappings
+    size_t page_ndx_in_file = local_page_ndx + m_first_page;
+    for (size_t i = 0; i < m_file.mappings.size(); ++i) {
+        EncryptedFileMapping* m = m_file.mappings[i];
+        if (m != this && m->contains_page(page_ndx_in_file)) {
+            size_t shadow_local_page_ndx = page_ndx_in_file - m->m_first_page;
+            if (is(m->m_page_state[shadow_local_page_ndx], UpToDate)) { // only keep up to data pages up to date
```

Comment:
What would lead to a page not being UpToDate with these changes? Is it only if the page has never actually been read yet?

Reply:
That, and the page-reclaimer may have reclaimed it.

Reply:
Perhaps add as comment?

Reply:
Given this and the former question, perhaps a comment is warranted?